### DISCUSSION TITLE
Error: Found at least one broken face in the mesh, see errors above for a more detailled analysis.

### DISCUSSION THREAD

**Discussion by buzhiqiuyuebai** [SeisSol/SeisSol#1029] [Q&A] [answered]
I'm using an[ example](https://github.com/buzhiqiuyuebai/RepeatmeshTest) test with two sets of meshes, they are completely overlapping geometrically, but are two separate sets of meshes, I'm getting this error, can the code support two overlapping sets of meshes please?
![image](https://github.com/SeisSol/SeisSol/assets/114474894/abf4bfe7-1f18-479d-919e-64fff43e0b84)

**Comment by sebwolf-de** [comment, marked as answer]
In theory, two overlapping meshes could work. Anyways, in your case the mesh is broken. You have several elements, which do not have a neighbor.

**Comment by buzhiqiuyuebai** [comment]
I've provided the example I'm running [here](https://github.com/buzhiqiuyuebai/RepeatmeshTest), it's a very simple cubic mesh, and I think the mesh is intact, but since it's two sets of meshes, the boundary conditions dictate that there are two triangles shared by neighboring tetrahedra, instead of a shared one, which I think is the cause of this error.

**Reply by sebwolf-de** [reply]
> I think the mesh is intact

No, it is not. Otherwise SeisSol wouldn't have thrown this error message.

**Reply by buzhiqiuyuebai** [reply]
"Do you know where the grid is not intact?"

**Reply by sebwolf-de** [reply]
SeisSol tells you the ids of the elements, which are broken.

**Comment by buzhiqiuyuebai** [comment]
> SeisSol tells you the ids of the elements, which are broken.

Can you please tell me how can I determine where the bad mesh is in the model, gmsh can see it in the GUI, how can file.puml.h5 locate where the mesh is in the model. Because I am again encountering this type of error reports.
![image](https://github.com/SeisSol/SeisSol/assets/114474894/0a2e1102-2be3-4a78-bb1d-75dacbb88f34)

**Reply by sebwolf-de** [reply]
The elements with the identifiers 25969360, 24834227, ... are broken.

**Reply by Thomas-Ulrich** [reply]
maybe we could output the barycenter of the element or face to facilitate debugging

**Reply by sebwolf-de** [reply]
This small python script
```
import seissolxdmf as sx
mesh = sx.seissolxdmf("mesh.xdmf")
conn = mesh.ReadConnect()
geom = mesh.ReadGeometry()
print(geom[conn[10]])
```
will give you the four corners of mesh element `10`:
```
array([[-17937.06299876,  37435.24717582, -18385.82170541],
       [-17752.1616131 ,  33080.52013952, -13801.35165624],
       [-14319.52030721,  35747.2608274 , -14982.39035261],
       [-19976.66074709,  36676.95003274, -10579.09563572]])
```
